### DISCUSSION TITLE
How to customize UI (view) of Traits

### DISCUSSION THREAD

**Discussion by stljeff1** [GrapesJS/grapesjs#3933] [Q&A]
I need to apply special styles to checkbox traits. I would also like to control the layouts of other traits based on trait type.

I need to target `.gjs-trt-trait` containers and apply classes there.

Is this possible? Can someone explain the purpose of the TraitViewer  class?

thanks

**Comment by artf** [comment]
Hi @stljeff1 the TraitViewer applies already its own classes by type so, you should be able to target them by those classes. In case you need more controlled customization the only way is to use [custom traits](https://grapesjs.com/docs/modules/Traits.html#define-new-trait-type).
